### ISSUE TITLE
feat(plugin-ledger-connector-burrow): add connector

### ISSUE THREAD

**Issue by petermetz** [hyperledger/cacti#292]
**Is your feature request related to a problem? Please describe.**

We need a connector to transact on Hyperledger Burrow.

**Describe the solution you'd like**

Following the pattern of the other connector plugins we have out there.

cc: @takeutak @sfuji822 @hartm @jonathan-m-hamilton @AzaharaC @jordigiam @kikoncuo @jagpreetsinghsasan @arnab-roy

**Comment by jscode017** [comment]
I would like to take this issue. But can I start working this maybe a week later? Kind of busy finishing my school final projects.

**Comment by petermetz** [comment]
@jscode017 Sure thing! It's been open for quite some time so it can definitely wait another week. :-)
I'll assign it to your name so that we know you are working on it!

**Comment by jscode017** [comment]
@petermetz I would probably start to work on this about Monday, after finish all my final projects at school. Would you recommend building cactus-plugin-ledger-connector-burrow first for a more test-driven fashion, or just write plugin-ledger-connector-burrow?

**Comment by petermetz** [comment]
@jscode017 That's cool, no rush at all!

My personal take on TDD specific to this situation: 
It depends on how familiar are you with the APIs of Burrow specifically. If you are fairly experienced with it and therefore know in advance what the tests should look like, then yeah, I think TDD is great. On the other hand if you are new to Burrow or not super familiar with it yet, then trying to define the tests up front will be difficult IMO because then the task of writing the connector will also be part research.
Strict TDD in my opinion works best when you are building something that you had built before like a run of the mill business application that is largely doing the same thing over and over again with a little twist. In scenarios like that it's fairly easy to know how a test will look like, but developing a new feature for a framework that is more on the research side in my opinion.

TLDR: I'd copy paste an existing connector's packages (both the test and the actual package) and then rename them accordingly and slowly work it over to have it operate on Burrow not the OG ledger. (this is what I usually do when writing new connectors)

**Comment by jscode017** [comment]
Thanks for the advice, copy-paste an existing connector's packages sounds like a great approach

**Comment by jscode017** [comment]
@petermetz, to clarify, what we need is to send transactions from cactus to burrow.
So I should focus on the client side of burrow like [this](https://github.com/hyperledger/burrow/blob/main/rpc/client.go)
and not something on burrow server side like [this](https://github.com/hyperledger/burrow/blob/main/protobuf/rpctransact.proto) right?
Please correct me if I am wrong.

**Comment by petermetz** [comment]
@jscode017 I agree that simpler version would be a better way to start just because it's easier to get going with it and see how things are. Writing a connector is at least partially a research task unless you have extensive prior experience with the underlying ledger so there's always a little bit of trial and error (at least there was for me when I started writing our connectors initially).
Anyway, I agree the client side abstraction is good to get started with and if you have any other questions just hit me up as usual.

**Comment by petermetz** [comment]
@jscode017 De-assigning it for now, please let me know if you'd like to pick this back up again!
